### PR TITLE
[linkis on k8s]Dev 0.10.0

### PR DESCRIPTION
**Pull request by brianzhangrong**
linkis on k8s
1.add  startup.sh,stop.sh,dockerfile  for  docker build
2.add k8s deployment,service(gateway NodePort),configmap,docker register-secret
3.test case 
[系统测试报告模版.docx](https://github.com/WeBankFinTech/Linkis/files/4796814/default.docx)

### CODE REVIEW

File: bml/bmlserver/pom.xml [hunk 1]
```diff
@@ -87,6 +112,26 @@
...
+                        <id>default</id>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>build</goal>
+                            <goal>tag</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <repository>10.206.0.5/library/linkis</repository>
```
[the remark this answers is not in the document]
Comment (brianzhangrong) [reply bml/bmlserver/pom.xml:130]: accept,use maven properties  to replace

[the remark this answers is not in the document]
Comment (brianzhangrong) [reply bml/bmlserver/pom.xml:130]: it's ok to remove 'resourceManager/resourcemanagerserver/dockerinfo/image'
`creationTimestamp: "2020-05-29 t07:58:40 Z"`

[the remark this answers is not in the document]
Comment (Adamyuanyuan) [reply bml/bmlserver/pom.xml]: Could we also remove private harbor address from dockerfile. thx

**Comment by Adamyuanyuan** [comment]
Could we also remove private harbor address from dockerfile. thx

**Comment by brianzhangrong** [comment]
> Could we also remove private harbor address from dockerfile. thx

we have plan to use public docker hub replace,it will be fix in next version

**Comment by brianzhangrong** [comment]
> 1、The Kubernetes deployment does not contain Eureka, Could we add eureka deployment and use the service dns name as the eurka url？
> 2、When I use kubectl to get log from pod, log message only show service start successfully . And I need to enter container to get whole log message.
> 3、Dockerfiles contains one line "COPY target/linkis-xx.zip /opt/ihome", I propose to add more instructions of this line in user guide.
> 4、The HDFS and Spark Config file add directly in docker images. when the cluster need to expand, the docker image need to rebuild.
> 5、How does the spark driver connect to cluster in kubernetes's pod? Could you provide more instructions in the user document?
---
1.it is ok to use fqdn to find eureka service,but  we consider use nacos to replace erueka.  Config Server and  Service Registry to use one system
2.now we mount log to host in path /var/log. You can use fluentd-es or EFK to collector log.
3.this is used for maven-docker-plugin, if  CI/CD  contains docker  build/tag/push  pipeline ,just  modify target zip path
4.k8s can mount  hostpath  to  replace hive/hadoop config file in docker container  
5. Yes. We use static route  and ip masque to receive packet  from spark executor.We will  add this detail in document

**Comment by peacewong** [comment]
Because too many commits were introduced, this PR needs to be resubmitted

**Pull request closed without merging**